### DISCUSSION TITLE
MessageDialog with multiple button and commands

### DISCUSSION THREAD

**Discussion by sschroedel** [israel-dryer/ttkbootstrap#413] [Q&A] [answered]
I currently migrating all my gui based py to ttkbootstrap as this is the best gui style and overall handling I experienced for Python. 

I'm struggling with having two buttons within the MessageBox and exdecuting different commands for each button.  Docstring says:

```
>   buttons (List[str]):
                A list of buttons to appear at the bottom of the popup
                messagebox. The buttons can be a list of strings which
                will define the symbolic name and the button text.
                `['OK', 'Cancel']`. Alternatively, you can assign a
                bootstyle to each button by using the colon to separate the
                button text and the bootstyle. If no colon is found, then
                the style is set to 'primary' by default.
                `['OK:success','Cancel:danger']`.
```

>             command (Tuple[Callable, str]):
>                 The function to invoke when the user closes the dialog.
>                 The actual command is a tuple that consists of the
>                 function to call and the symbolic name of the button that
>                 closes the dialog.

What I understood or tried is:

```
mdDel = MessageDialog(title="Confirm", message="I want to delete all my data",
buttons=["YES, I do:danger", "No, abbort:success"],
command=[(yay, "OK"), (nay, "NOK")],
alert=True)
mdDel.show()
```

I tried several thing with lambda etc. I currently thinking about just builidng my own message but maybe there is hope and somebody can help me.

Cheers

**Comment by sschroedel** [comment]
Anyone?

**Reply by FranzKol** [reply]
it's pretty simple, you can evaluate the result of the message dialog.
In **mdDel.result** you get the text of the key which was pressed when the message dialog was closed.
In your case the result will be either "**YES, I do**" or "**No, abbort**"

Regards

**Reply by sschroedel** [reply]
Thank you FranzKol - this is not what I intended but still a working bypass :)

**Comment by israel-dryer** [comment, marked as answer]
@FranzKol is right, and it's actually the intended API rather than a bypass. `MessageDialog` doesn't support a per-button command list — `command=` is a **single** callback fired on *any* button press (the old docstring's `Tuple[Callable, str]` wording was misleading and has since been corrected). To branch on *which* button was pressed, read `dlg.result`, which holds the text of the button that closed the dialog:

```python
dlg = MessageDialog(
    title="Confirm",
    message="Delete all my data?",
    buttons=["Yes:danger", "No:success"],
)
dlg.show()
if dlg.result == "Yes":
    yay()
elif dlg.result == "No":
    nay()
```

`result` is the button *text* (so `"Yes:danger"` yields `"Yes"`; it's the translated text when `localize=True`). Since `result` is set *before* the `command` callback runs, you can alternatively pass a single `command` that dispatches on `dlg.result` — but the post-`show()` branch above is the simplest.

**2.0 note:** the signature is cleaned up to a plain zero-arg `Callable`, and the legacy `(callable, label)` tuple form (what you originally tried) is explicitly deprecated — it now emits a `DeprecationWarning` and unwraps to the callable, with removal in 3.0. The `result`-branching pattern above stays the same.
